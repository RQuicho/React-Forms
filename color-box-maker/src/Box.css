.Box-btn {
  display: flex;
  justify-content: space-between;
  padding: 10px;
  margin: 5px;
}