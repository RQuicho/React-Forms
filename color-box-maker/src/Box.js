import {React, useState} from 'react';
import './Box.css';

const Box = ({color, width, height}) => {
  const [isVisible, setIsVisible] = useState(true);
  const remove = () => {
    setIsVisible(false);
  }
  return (
    <>
    {isVisible && 
      <div className='Box' style={{backgroundColor: color, width: width, height: height}}>
        <p>{color}</p>
      </div>  
    }
    {isVisible &&
      <button className='Box-btn' onClick={remove}>X</button>
    }
    </>
  )
}

export default Box;