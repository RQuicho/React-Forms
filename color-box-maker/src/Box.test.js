import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import Box from './Box';

// smoke test
it('renders without crashing', () => {
  render (<Box color='red' width={150} height={150}/>);
});

// snapshot test
it('matches snapshot', () => {
  const {asFragment} = render(<Box color='red' width={150} height={150}/>);
  expect(asFragment()).toMatchSnapshot();
});

it('should remove box and button when X btn is clicked', () => {
  const {queryByText} = render(<Box color='red' width={150} height={150}/>);
  const btn = queryByText("X");
  expect(queryByText('red')).toBeInTheDocument(); // red and blue box are initial state
  fireEvent.click(btn); // removes box
  expect(queryByText('red')).not.toBeInTheDocument();
});