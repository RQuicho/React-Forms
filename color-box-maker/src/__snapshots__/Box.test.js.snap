// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: red; width: 150px; height: 150px;"
  >
    <p>
      red
    </p>
  </div>
  <button>
    X
  </button>
</DocumentFragment>
`;
